### PR TITLE
OCPBUGS-42067: Reapply "chore: poll immediately in the e2e tests"

### PR DESCRIPTION
**Pull request by simonpasquier**
This reverts commit 10160fd3fefb3791121ad8f5b8bf043938085884.

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -655,7 +655,7 @@ func (f *Framework) ForwardPort(t *testing.T, ns, svc string, port int) (string,
 func Poll(interval, timeout time.Duration, f func() error) error {
 	var lastErr error
 
-	err := wait.PollUntilContextTimeout(context.Background(), interval, timeout, false, func(context.Context) (bool, error) {
```
Comment (machine424) [inline test/e2e/framework/framework.go:658]: All of the runs I checked were timing out, I think this is the change that may make a difference. Do you think we need to have the rest as well? (as I'm not seeing any flaky tests)

**Comment by machine424** [comment]
Discussed offline, let's give this a try, thanks.
/lgtm

**Comment by simonpasquier** [comment]
From the e2e-agnostic-operator job run

```
Step phase test succeeded after 1h33m48s
````

**Pull request merged**